app/models/order.rb

```diff
@@ -42,6 +47,20 @@ def already_has_product?(product)
     products.include? product
   end
 
+  def buyer_card_unexpired
+    # guard clause
+    return if pending? || confirmed_payment
```

Comment:
I'd delete `pending?` here -- you already have that filter above. This method is not called if the status is pending.